pkg/graph/resolvers/plan_document.go

```diff
@@ -0,0 +1,80 @@
+package resolvers
+
+import (
+	"github.com/cmsgov/mint-app/pkg/graph/model"
+	"github.com/cmsgov/mint-app/pkg/models"
+	"github.com/cmsgov/mint-app/pkg/storage"
+	"github.com/cmsgov/mint-app/pkg/storage/genericmodel"
+	"github.com/cmsgov/mint-app/pkg/upload"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+func createDocumentPayload(s3Client *upload.S3Client, document *models.PlanDocument) (*model.PlanDocumentPayload, error) {
+	presignedURL, urlErr := s3Client.NewGetPresignedURL(*document.FileKey)
+	if urlErr != nil {
+		return nil, urlErr
+	}
+
+	payload := model.PlanDocumentPayload{
+		Document:     document,
+		PresignedURL: presignedURL,
+	}
+
+	return &payload, nil
+}
+
+// PlanDocumentCreate implements resolver logic to create a plan document object
+func PlanDocumentCreate(logger *zap.Logger, input *model.PlanDocumentInput, principal *string, store *storage.Store, s3Client *upload.S3Client) (*model.PlanDocumentPayload, error) {
+	document, err := store.PlanDocumentCreate(logger, principal, input, s3Client)
+	if err != nil {
+		return nil, genericmodel.HandleModelUpdateError(logger, err, models.PlanDocument{ID: *input.ID})
+	}
+
+	return createDocumentPayload(s3Client, document)
+}
+
+// PlanDocumentRead implements resolver logic to fetch a plan document object by ID
+func PlanDocumentRead(logger *zap.Logger, id uuid.UUID, store *storage.Store) (*models.PlanDocument, error) {
```

Comment:
I think `get` would be more consistent in this spot as well.

Reply:
@ClayBenson94 Let's have a discussion on naming conventions. I went with Read as it matches the CRUD acronym, but Steven raises a good point and we should align one way or another.